test/distributions/test_distributions.py

```diff
@@ -2773,6 +2773,20 @@ def test_dirichlet_log_prob(self):
             expected_log_prob = scipy.stats.dirichlet.logpdf(x[i].numpy(), alpha.numpy())
             self.assertEqual(actual_log_prob[i], expected_log_prob, atol=1e-3, rtol=0)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_dirichlet_log_prob_zero(self):
+        # Specifically test the special case where x=0 and α=1.  The PDF is
+        # proportional to x**(α-1), which in this case works out to 0**0=1.
+        # The log PDF of this term should therefore be 0.  However, it's easy
+        # to accidentally introduce NaNs by calculating log(x) without regard
+        # for the value of α-1.
+        alpha = torch.tensor([1, 2])
+        dist = Dirichlet(alpha)
+        x = torch.tensor([0, 1])
+        actual_log_prob = dist.log_prob(x)
+        expected_log_prob = scipy.stats.dirichlet.logpdf(x.numpy(), alpha.numpy())
+        self.assertEqual(actual_log_prob, expected_log_prob, atol=1e-3, rtol=0)
```

Comment:
Do you really need to override the tolerances here? I would expect the default one to work fine here?

Reply:
I just copied the whole assertion line from the existing `Dirichlet.log_prob()` test.  I don't know if there's a good reason why these tolerances were chosen in the first place, though.  Maybe there's a test environment that uses low-precision floats or something?  In any case, my goal was to be consistent.  Also, this test only really needs to distinguish between NaN and not NaN, so for that the tolerance doesn't matter.